actions/pull-request/download-artifact/entrypoint/main.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/paketo-buildpacks/github-config/actions/pull-request/download-artifact/entrypoint/internal"
+)
+
+func main() {
+	var options internal.Options
+
+	flag.StringVar(&options.Name, "name", "", "Name of the uploaded artifact")
+	flag.StringVar(&options.Repo, "repo", "", "Org and repository that the workflow lives in")
+	flag.StringVar(&options.RunID, "runID", "", "ID of the specific workflow that contains the artifact")
+	flag.StringVar(&options.GithubAPI, "githubAPI", "", "Github API endpoint to query for the download")
+	flag.Parse()
+
+	if options.Name == "" {
+		fail(errors.New(`missing required input "name"`))
+	}
+
+	if options.Repo == "" {
+		fail(errors.New(`missing required input "repo"`))
+	}
+
+	if options.RunID == "" {
+		fail(errors.New(`missing required input "runID"`))
+	}
+
+	archiveDownloadURL, zipSize, err := internal.GetWorkflowArtifactURL(options, os.Getenv("GITHUB_TOKEN"))
+	if err != nil {
+		fail(err)
+	}
+
+	fmt.Println("Getting workflow artifact zip file")
+	if options.GithubAPI != "https://api.github.com" {
+		archiveDownloadURL = options.GithubAPI + archiveDownloadURL
+	}
+
+	payloadBody, err := internal.GetArtifactZip(archiveDownloadURL, os.Getenv("GITHUB_TOKEN"))
+	if err != nil {
+		fail(err)
+	}
+
+	unzippedFileBytes, err := internal.UnzipPayload(options.Name, payloadBody, zipSize)
+	if err != nil {
+		fail(err)
+	}
```

Comment:
Yea, I definitely felt a bit strange about how this code is currently working but I wasn't sure how to mitigate it. Your idea sounds reasonable, I'll try it out now.

Reply:
Let me know if this is what you had in mind: https://github.com/paketo-buildpacks/github-config/pull/267/commits/50000317ff29f09c47e09473b8e11eee5e133983
I tried to find a different way to iterate through the zip file but I couldn't land on any alternatives.

Reply:
I think what you did is really close to what I had in mind. I made some small changes in https://github.com/paketo-buildpacks/github-config/pull/267/commits/45d0478618f182fcccbfcdc5728753ce7bd5783b to get it all the way there.

Reply:
Ah clever, I'm good with this. I see now that `io.Copy` is a more memory efficient option than doing the `os.ReadAll` because of the fixed buffer.  Thanks for fixing that